featuregates/client/pkg/util/featuregate.go

```diff
@@ -131,3 +140,71 @@ func FeaturesActivatedInNamespacesMatchingSelector(ctx context.Context, c client
 	}
 	return true, nil
 }
+
+// IsFeatureActivated returns true only if the feature is activated.
+func IsFeatureActivated(ctx context.Context, c client.Client, featureName string) (bool, error) {
+	feature := &corev1alpha2.Feature{}
+	if err := c.Get(ctx, types.NamespacedName{
+		Name: featureName,
+	}, feature); err != nil {
+		return false, fmt.Errorf("couldn't retrieve feature %s :%w", featureName, err)
+	}
+
+	return feature.Status.Activated, nil
+}
+
+// GetFeatureGateForFeature returns FeatureGate resource that is gating the feature
+func GetFeatureGateForFeature(ctx context.Context, c client.Client, featureName string) (*corev1alpha2.FeatureGate, bool, error) {
+	featureGateList := &corev1alpha2.FeatureGateList{}
+	if err := c.List(ctx, featureGateList); err != nil {
+		return nil, false, fmt.Errorf("couldn't list FeatureGate resources: %w", err)
+	}
+
+	for i, fg := range featureGateList.Items {
+		for _, f := range featureGateList.Items[i].Spec.Features {
+			if f.Name == featureName {
+				featureGate := &corev1alpha2.FeatureGate{}
+				if err := c.Get(ctx, types.NamespacedName{
+					Name: fg.Name,
+				}, featureGate); err != nil {
+					return nil, false, fmt.Errorf("couldn't get FeatureGate resource %s: %w", featureGate.Name, err)
```

Comment:
I feel like I'm missing something obvious, but why not just `return fg, true, nil`?